inst/scripts/new_post.R

```diff
@@ -14,11 +14,11 @@ xfun::in_dir(blogdown:::site_root(), local({
     miniUI::miniPage(miniUI::miniContentPanel(
       txt_input('title', 'Title', placeholder = 'Post Title'),
       shiny::fillRow(
-        txt_input('author', 'Author', getOption('blogdown.author', ''), width = '98%'),
+        txt_input('author', 'Author', blogdown:::get_author(), width = '98%'),
         shiny::dateInput('date', 'Date', Sys.Date(), width = '98%'),
-        txt_input(
-          'subdir', 'Subdirectory', getOption('blogdown.subdir', 'post'),
-          '(optional)', width = '98%'
+        shiny::selectInput(
+          'subdir', 'Subdirectory', blogdown:::get_dir(),
+          width = '98%'
         ),
```

Comment:
```suggestion
        shiny::selectizeInput(
          'subdir', 'Subdirectory', blogdown:::get_dir(),
          width = '98%', multiple = FALSE, options = list(create = TRUE)
        ),
```

One small suggestion: if you use `shiny::selectizeInput()` with `options = list(create = TRUE)`, then we can retain the best of both words by allowing users to create a new folder or pick from existing folders.

If you make this change, you might also want to consider more broadly changing the behavior of `get_dir()` so that the full list of directories under `content/` is shown as `choices` but the default `selected` is `getOption(blogdown.subdir)` or `'post'`, etc.

Reply:
Fantastic idea, I'll do that tomorrow or later this week!